apps/condo/domains/resident/utils/helpers.js

```diff
@@ -0,0 +1,68 @@
+const get = require('lodash/get')
+const dayjs = require('dayjs')
+const isSameOrBefore = require('dayjs/plugin/isSameOrBefore')
+const { NOT_FOUND_ERROR } = require('@condo/domains/common/constants/errors')
+
+dayjs.extend(isSameOrBefore)
+
+/**
+ * Connects or disconnects residents to/from organization & property.
+ * Property should be provided to connect.
+ * In other case residents will be disconnected.
+ * @param ResidentAPI (API) either Server or Test API
+ * @param context (context)
+ * @param residents (Resident[])
+ * @param property (Property)
+ * @param reconnectToOlder (boolean) forces reconnecting residents with non-deleted property to restored older one
+ * @returns {Promise<void>}
+ */
+const connectResidents = async (ResidentAPI, context, residents, property, reconnectToOlder = false) => {
+    // Nothing to connect
+    if (!Array.isArray(residents)) return
+    
+    const propertyId = get(property, 'id')
+    const shouldConnect = Boolean(property) && propertyId
+    const organizationId = shouldConnect ? property.organization.id : null
+    const organizationConnect = shouldConnect ? { connect: { id: organizationId } } : { disconnectAll: true }
+    const propertyConnect = shouldConnect ? { connect: { id: propertyId } } : { disconnectAll: true }
+
+    for (const resident of residents) {
+        const residentId = get(resident, 'id')
+
+        if (shouldConnect) {
+            // NOTE: for optimization sake, we decide here if property should be switched, by calculating which property is older, current or next one
```

Comment:
it's better to add resident query filter for that

Reply:
done in `manageResidentToPropertyAndOrganizationConnections` task